numba_rvsdg/core/transformations.py

```diff
@@ -298,22 +309,17 @@ def find_branch_regions(scfg: SCFG, begin: str, end: str) -> Set[str]:
     return branch_regions
 
 
-def _find_branch_regions(scfg: SCFG, begin: str, end: str) -> Set[str]:
-    # identify branch regions
-    branch_regions = []
-    for bra_start in scfg[begin].jump_targets:
-        region = []
-        region.append(bra_start)
-    return branch_regions
-
-
 def find_tail_blocks(
-    scfg: SCFG, begin: Set[str], head_region_blocks, branch_regions
-):
+    scfg: SCFG,
+    begin: str,
+    head_region_blocks: Set[str],
+    branch_regions: List[Optional[Tuple[str, Set[str]]]],
+) -> Set[str]:
     tail_subregion = {b for b in scfg.graph.keys()}
     tail_subregion.difference_update(head_region_blocks)
     for reg in branch_regions:
         if not reg:
```

Comment:
Maybe it's better to be explicit here? The placeholder is now `None` rather than an empty tuple. 

```suggestion
        if reg is None:
```